Update roland_cutstudio.py

vmario89 opened the pull request:
Added support for Roland CutStudio with Wine. Tested on Ubuntu 20 LTS 64 Bit. converted eps from temp dir successfully imports into Cutstudio

Some notes for this:
you need to run winecfg to add another virtual drive "T:" which points to "/tmp/" of the host computer. Otherwise CutStudio will not find the file

And there are some left deprecation warnings. I will file an issue for this because i dont know how to get rid of them yet

vmario89 committed:
Update roland_cutstudio.py

Added support for Roland CutStudio with Wine. Tested on Ubuntu 20 LTS 64 Bit. converted eps from temp dir successfully imports into Cutstudio

Some notes for this:
you need to run winecfg to add another virtual drive "T:" which points to "/tmp/" of the host computer. Otherwise CutStudio will not find the file

And there are some left deprecation warnings. I will file an issue for this because i dont know how to get rid of them yet

mgmax commented:
I like the idea. Could you check the return status of Wine and if it fails, print some setup instructions and fall back to the previous behavior? Then, someone who has installed Wine but has not installed CutStudio or not configured the appropriate directory would get an information what they should do.

By the way, does printing from CutStudio in Wine actually work, especially the "Get size from machine" command?

mgmax committed:
Fix deprecation warnings (#11), add selftest

vmario89 committed:
Added some more convenience with wine

vmario89 commented:
Hi, i added some wrapper to check if wine exists. if yes then it will check if cutstudio was configured. if not it will just behave like before =) i dont know if cutstudio also cuts with our machine in linux - had no time to test yet but i will try next days/weeks. we have an old roland pnc 950 machine which actually only runs in a virtual machine with microsoft Losedos XP. but i often use inkscape to just export to roland CS and to save the cst files. this helps me a lot preparing cutting jobs =)

mgmax commented on roland_cutstudio.py:
```diff
@@ -42,6 +41,8 @@
 import shutil
 import numpy
 from functools import reduce
+import inkex
```
I'd suggest not using `inkex` because then there is no easy way to test the script outside Inkscape. `print("...", file=sys.stderr)` also works.

mgmax commented on roland_cutstudio.py:
```diff
@@ -388,21 +393,28 @@ def output(array):
+    Popen([which("CutStudio\CutStudio.exe", True), "/import", filename+".cutstudio.eps"], creationflags=DETACHED_PROCESS, close_fds=True)
+else: #check if we have access to "wine"
+    if which("wine", False) is not None:
+        if which("/opst/CutStudio/CutStudio.exe", False) is not None:
```
typo: `/opst/` -> `/opt/`

mgmax commented on roland_cutstudio.py:
```diff
@@ -388,21 +393,28 @@ def output(array):
+            with os.popen("wine /opt/CutStudio/CutStudio.exe /import T:\\\\" + filename.split('/tmp/')[1] + ".cutstudio.eps", "r") as cutstudio:
+                result = cutstudio.read()
+        else:
+            inkex.utils.debug("Found a wine installation on your system but no CutStudio.exe. You can easily emulate this Windows application on Linux using wine. To do this provide a valid CutStudio installation in directory \"/opt/CutStudio/\". You may create a symlink with the shell command \"ln -sf /your/path/to/cutstudio /opt/CutStudio\" to point to this dir, or just edit the InkScape extens ...
```
- Maybe flip the message: First print the filename, then the long text?
- Maybe also show it on Linux if wine is not installed, since it's easy to install?
- Why `/opt`? It is usually not user-writable, so you'll have to create the symlink as root. We could simply use the default install directory: `$HOME/.wine/dosdevices/c:/Progr*/CutStudio/CutStudio.exe` Also, this path should be *one* variable, not two places in the script.
- For extra comfort, we could also copy the file to `$HOME/.wine/dosdevices/c:/temp.cutstudio.eps` or similar. So the T: drive is not required.

mgmax reviewed (changes requested):
Thanks! I'd like to suggest some small changes that would make it a lot better.

vmario89 committed:
Wine adjustments

mgmax committed:
Merge https://github.com/vmario89/inkscape-roland-cutstudio into wine

mgmax committed:
code cleanup

mgmax commented:
I cleaned it up a little and changed the message. Can you please check if everything still works fine with WINE?

vmario89 committed:
Update roland_cutstudio.py

Hi. Tested your code. After adusting the two lines it perfectly works with wine and with zero error or deprecation output. Good job Max. Thanks again for your brain-sharing =)

vmario89 reviewed (commented):
Hi. Did minimal adjustment with some variable issue related to Wine. Now we can drink a red wine :-D
